File: prueba-tecnica-backend/src/main/java/com/lautaro_cenizo/service/VehiculoService.java
package com.lautaro_cenizo.service;

import com.lautaro_cenizo.crud.dto.VehiculoDTO;
import com.lautaro_cenizo.entity.vehiculo.Vehiculo;

import java.util.List;

public interface VehiculoService {
    Vehiculo crear(VehiculoDTO vehiculoDTO);
    Vehiculo actualizar(Integer id, VehiculoDTO vehiculoDTO);
    Vehiculo obtenerPorId(Integer id);
    Vehiculo obtenerPorPatente(String patente);
    List<Vehiculo> listarTodos();
    List<Vehiculo> listarPorCliente(Integer clienteId);
    void eliminar(Integer id);
    void cambiarCliente(Integer vehiculoId, Integer nuevoClienteId);
}

File: prueba-tecnica-frontend/src/app/service/vehiculo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Vehiculo } from '../models/class/vehiculo';
import { VehiculoDTO } from '../models/dto/vehiculoDto';

@Injectable({
  providedIn: 'root'
})
export class VehiculoService {

  private baseUrl = 'http://localhost:8080/api/vehiculos';

  constructor(private http: HttpClient) { }

  crear(vehiculoDTO: VehiculoDTO): Observable<Vehiculo> {
    return this.http.post<Vehiculo>(this.baseUrl, vehiculoDTO);
  }

  actualizar(id: number, vehiculoDTO: VehiculoDTO): Observable<Vehiculo> {
    return this.http.put<Vehiculo>(`${this.baseUrl}/${id}`, vehiculoDTO);
  }

  obtenerPorId(id: number): Observable<Vehiculo> {
    return this.http.get<Vehiculo>(`${this.baseUrl}/${id}`);
  }

  obtenerPorPatente(patente: string): Observable<Vehiculo> {
    return this.http.get<Vehiculo>(`${this.baseUrl}/patente/${patente}`);
  }

  listarTodos(): Observable<Vehiculo[]> {
    return this.http.get<Vehiculo[]>(this.baseUrl);
  }

  listarPorCliente(clienteId: number): Observable<Vehiculo[]> {
    return this.http.get<Vehiculo[]>(`${this.baseUrl}/cliente/${clienteId}`);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  cambiarCliente(vehiculoId: number, nuevoClienteId: number): Observable<void> {
    return this.http.put<void>(
      `${this.baseUrl}/${vehiculoId}/cambiar-cliente/${nuevoClienteId}`, 
      null
    );
  }
}
